app/src/main/java/com/example/barterbuddy/activities/ChatPage.java

```diff
@@ -61,21 +55,26 @@ protected void onCreate(Bundle savedInstanceState) {
 
     currentUserId = AuthenticationUtil.getCurrentUserEmail();
     otherUserEmail = getIntent().getStringExtra("otherUserEmail");
-    assert otherUserEmail != null;
-    chatroomId = FirebaseUtil.getChatroomId(currentUserId, otherUserEmail);
-    tradeId = FirebaseUtil.getTradeId(currentUserId, otherUserEmail);
+    userIsPoster = getIntent().getBooleanExtra("isPoster", false);
+    if (userIsPoster) {
+      chatroomId = currentUserId + "_" + otherUserEmail;
```

Comment:
I'm going to delete them, let me know if you'd prefer otherwise. I think because it is such a simple action (chats and trades just have posterEmail_offeringEmail as their ID) they don't need methods, unless for code readability.

Reply:
Fine by me, it's simple enough to do outside of a method.